manta-pay/abstract-protocol.tex

```diff
@@ -59,14 +88,60 @@ \subsection{Abstract Cryptographic Schemes}
     \end{itemize}
 \end{definitiontoc}
 
+    \textbf{Notation}: For convenience, we may refer to $\KA.\agree(\sk, \textsf{D})$ as $\KA.\agree_{\textsf{D}}(\sk)$ for all $\sk : \SecretKey$ and $\textsf{D} : \PublicKey$.
+
+\begin{definitiontoc}{Message Authentication Code}
+    A \emph{message authentication code} $\MAC$ is given by the schema:
+    \begin{align*}
+        \SecretKey &: \Type \\
+        \Message   &: \Type \\
+        \Tag       &: \Type \\
+        \sign      &: \SecretKey \times \Message \to \Tag \\
+        \verify    &: \SecretKey \times \Message \times \Tag \to \Bool
+    \end{align*}
+    with the following properties:
+    \begin{itemize}
+        \item \textbf{Completeness}: For all $\sk : \SecretKey$, $m : \Message$, we have that $\verify(\sk, m, \sign(\sk, m)) = \True$.
+        \item \textbf{Unforgeability}: For any key $\sk : \SecretKey$ and efficient adversary $\mathcal{A}_\sk$ with oracle access to $\sign(\sk, \dash)$ the following probability is negligible:
+            \[
+                \Prob{
+                    \verify(\sk, m, t) = \True
+                }{
+                    (m, t) \sim \mathcal{A}_\sk \\
+                    m \not\in \mathcal{Q}(\mathcal{A}_\sk)
+                }
+            \]
+            where $\mathcal{Q}(\mathcal{A}_\sk)$ denotes the set of queries that $\mathcal{A}_\sk$ makes to the oracle during its strategy.
+    \end{itemize}
+
+    \textbf{Note}: In the case that the secret key should not be known by the verifier, one can use a signature scheme instead.
+\end{definitiontoc}
+
+\begin{definitiontoc}{Signature Scheme}
+    A \emph{signature scheme} $\SIG$ is defined by the schema:
+    \begin{align*}
+        \SecretKey    &: \Type \\
+        \PublicKey    &: \Type \\
+        \Message      &: \Type \\
+        \Signature    &: \Type \\
+        \derive       &: \SecretKey \to \PublicKey \\
+        \sign         &: \SecretKey \times \Message \to \mathfrak{D}\bra{\Signature} \\
```

Comment:
The meaning of $\mathfrak{D}\bra{\Signature}$ is unclear.

Reply:
@bhgomes Remove distribution from traits, but they are still needed for security properties.